pkg/cmd/infra/router/template.go

```diff
@@ -565,3 +566,88 @@ func (o *TemplateRouterOptions) blueprintRoutes(routeclient *routeclientset.Clie
 
 	return blueprints, nil
 }
+
+// makeTLSConfig checks whether metrics TLS is configured and
+// if so returns a tls.Config whose certificate is automatically
+// reloaded every 5 minutes or in response to file changes.
+func makeTLSConfig() (*tls.Config, error) {
+	certFile := env("ROUTER_METRICS_TLS_CERT_FILE", "")
+	if len(certFile) == 0 {
+		return nil, nil
+	}
+	keyFile := env("ROUTER_METRICS_TLS_KEY_FILE", "")
+
+	// Make sure we have something useful to start with
+	certificate, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		return nil, err
+	}
+
+	// Safely reload the certificate as requested.
+	var lock sync.Mutex
```

Comment:
Might as well use `sync.RWMutex`.

Reply:
Bikeshed, but RWMutex has (I think) objectively slightly trickier semantics, and this seems like a low to almost-no contention code path... (contention would occur when a cert secret changed during a rotation at the instant prometheus opens a connection for its periodic scrape?)

Reply:
And even then, we're talking about resolving the contention while a couple of files are read from disk (likely mapped to a RAM disk)... so a really dumb/simple lock seems appropriate

Reply:
All right, yeah, since we have only the one reader, `sync.Mutex` is no worse than `sync.RWMutex`.  It seems like natural place to use `sync.RWMutex`, but I realize now doing so would not help anything.